harvester/exceptions.py

```diff
@@ -51,43 +53,38 @@ class SendNotificationException(HarvestCriticalException):
     pass
 
 
-# non-critical exceptions
-class HarvestNonCriticalException(Exception):
-    def __init__(self, msg, harvest_job_id, harvest_record_id):
-        super().__init__(msg, harvest_job_id, harvest_record_id)
+def log_non_critical_error(msg, job_id, record_id, error_type, emit_log=True):
+    """Log a non-critical error into the database and logs.
 
-        self.msg = msg
-        self.harvest_job_id = harvest_job_id
-        self.harvest_record_id = harvest_record_id
+    If emit_log is False, then don't print the error into our logs.
+    """
+    error_data = {
+        "message": msg,
+        "type": error_type,
+        "date_created": datetime.now(timezone.utc),
+        "harvest_job_id": job_id,
+        "harvest_record_id": record_id,
+    }
 
-        self.db_interface: HarvesterDBInterface = db_interface
-        self.logger = logging.getLogger("harvest_runner")
+    db_interface.add_harvest_record_error(error_data)
+    db_interface.update_harvest_record(record_id, {"status": "error"})
 
-        error_data = {
-            "message": self.msg,
-            "type": self.__class__.__name__,
-            "date_created": datetime.now(timezone.utc),
-            "harvest_record_id": harvest_record_id,  # to-do
-            "harvest_job_id": harvest_job_id,
-        }
+    if emit_log:
+        logger.error(msg, exc_info=True)
 
-        self.db_interface.add_harvest_record_error(error_data)
-        self.db_interface.update_harvest_record(harvest_record_id, {"status": "error"})
-        self.log_err()
 
-    def log_err(self):
-        self.logger.error(self.msg, exc_info=True)
+# non-critical exceptions
+class HarvestNonCriticalException(Exception):
+    def __init__(self, msg, harvest_job_id, harvest_record_id):
+        super().__init__(msg, harvest_job_id, harvest_record_id)
+        self.msg = msg
+        log_non_critical_error(msg, harvest_job_id, harvest_record_id, self.__class__.__name__)
 
 
 class ExternalRecordToClass(HarvestNonCriticalException):
     pass
 
 
-class ValidationException(HarvestNonCriticalException):
-    def log_err(self):
-        pass
-
-
```

Comment:
Why do we want to remove this entirely? Isn't there a scenario where the validation fails (ie there is no title) and we want the normal flow to occur (exception thrown, error logged in DB, record updated as error, and log printed)?